classes/webservice.php

```diff
@@ -412,13 +412,16 @@ public function list_users() {
      * Incrementally retrieves the active paid users and compares against $numlicenses.
      * @see $numlicenses
      * @return bool Whether the paid user license limit has been reached.
+     * @throws dml_exception
      */
     private function paid_user_limit_reached() {
         $userslist = $this->list_users();
         $numusers = 0;
         foreach ($userslist as $user) {
-            if ($user->type != ZOOM_USER_TYPE_BASIC && ++$numusers >= $this->numlicenses) {
-                return true;
+            if (core_user::get_user_by_email($user->email)) {
+                if ($user->type != ZOOM_USER_TYPE_BASIC && ++$numusers >= $this->numlicenses) {
```

Comment:
Only use a database lookup for users who might be relevant. Move the `$numusers` increment operation outside of the conditional.
```suggestion
            if ($user->type != ZOOM_USER_TYPE_BASIC && core_user::get_user_by_email($user->email)) {
                $numusers++;
                if ($numusers >= $this->numlicenses) {
```